### PR TITLE
[FIX] website: missing alternate URLs for pages

### PR DESCRIPTION
**Pull request by Julien00859**
Install multiple langages, each time translating the website. In a
private browsing session access /contactus, show the page source, the
multiple alternate URL (`<link rel="alternative">` in the `<head>`) are
all pointing the canonical URL instead of the alternative.

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by rdeodoo — CHANGES_REQUESTED** [review]
Thanks for the quick PR and for the tests :+1: :+1:

**Review by rdeodoo — APPROVED** [review]
@robodoo r+
Thank you @Julien00859 

I tested every flow I could think of, homepage, controller, pages. With 3 langs, everything seems fine.

About the special case where one would select a specific page as homepage (eg `/my-home-page` being served on `/`), the change here about the canonical tag is ok. The canonical on `/` will now be `/my-home-page` and not `/` anymore.

I am not 100% sure what is the best solution, as in that case `/my-home-page` is still accessible through that URL, it means both `/my-home-page` and `/` will serve the same page (and so the same duplicate content).

Both URL having their own canonical despite being duplicated content seems weird.
The change here about both having `/my-home-page` as canonical for both URL seems logical but not ideal. I guess `/` on both would be better but that would require a special check/code when serving a page which is set as homepage with a URL != than `/` to "hack" the canonical tag to be `/`.

That would need some extra reading to find the best way to handle this special use case, which will do later as the fix here is really more important to handle.

**tl;dr**: with a page `/my-home-page` being set as homepage, it will be rendered when accessing `/`, but can still be accessed through `/my-home-page`.
- Before this PR: The canonical would be `/my-home-page` on `/my-home-page` and `/` on `/`, despite being the same duplicate content
- Now: The canonical will be `/my-home-page` on both pages
- Ideal later (to confirm): The canonical should be `/` on both pages. (And/or maybe accessing `/my-home-page` should redirect to `/`)

**Comment by rdeodoo** [comment]
For the record, the bug fixed here is the following, see the lang URL part is not correct

For the record, saas-15.2:
image

Saas-15.3:
image
